Tasks/AzureRmWebAppDeployment/KuduUtility.ts

```diff
@@ -0,0 +1,89 @@
+import Q = require('q');
+import tl = require('vsts-task-lib/task');
+import path = require("path");
+import fs = require("fs");
+var request = require ('request');
+var archiver = require('archiver');
+
+
+/**
+ *  Deploys a zip based webapp package.
+ * 
+ *  @param  webAppPackage                  Zip file or folder for deployment
+ *  @param  virtualApplicationMappings     Mapping to get physical path for deployment
+ *  @param  publishingProfile              publish profile provides destination details for deployment
+ *  @param  virtualApplication             (Optional) Virtual application name 
+ */
+export async function deployWebAppPackage(webAppPackage: string, virtualApplicationMappings, publishingProfile,
+                             virtualApplication: string) {
+
+    var deferred = Q.defer<any>();	
```

Comment:
You're inside an async function, no need for all this defer and reject promise code.